src/coreclr/jit/lower.cpp

```diff
@@ -2198,6 +2210,124 @@ GenTree* Lowering::LowerTailCallViaJitHelper(GenTreeCall* call, GenTree* callTar
     return result;
 }
 
+//------------------------------------------------------------------------
+// LowerCFGCall: Potentially lower a call to use control-flow guard. This
+// expands indirect calls into either a validate+call sequence or to a dispatch
+// helper taking the original target in a special register.
+//
+// Arguments:
+//    call         -  The call node
+//
+void Lowering::LowerGreenThreadTransitionCall(GenTreeCall* call)
+{
+    if (call->IsHelperCall(comp, CORINFO_HELP_COMPUTE_GREEN_THREAD_TRANSITION))
+    {
+        return;
+    }
+
+    GenTree* callTarget = call->gtCallType == CT_INDIRECT ? call->gtCallAddr : call->gtControlExpr;
+    if (callTarget == nullptr)
+    {
+        assert(false); // This should never happen for a p/invoke
+    }
+    else
+    {
+// Unlike CFG we need to handle integral constants
+#if 0
+        if (callTarget->IsIntegralConst())
+        {
+            // This is a direct call, no CFG check is necessary.
+            return;
+        }
+#endif
+    }
```

Comment:
It's ok with me if you remove the parallels to CFG here (assert not null, delete the else)

Reply:
Eh, it'll work for now.